autoprof/image/image_object.py

```diff
@@ -107,40 +136,60 @@ def shape(self):
         """
         return self.window.shape
     @property
-    def center(self):
+    def center(self)-> torch.Tensor:
         """
         Returns the center of the image window.
 
         Returns:
             torch.Tensor: A 1D tensor of shape (2,) containing the (x, y) coordinates of the center.
         """
         return self.window.center
-    def center_alignment(self):
-        """Determine if the center of the image is aligned at a pixel center
-        (True) or if it is aligned at a pixel edge (False).
+    def center_alignment(self)-> torch.Tensor:
+        """Determine if the center of the image is aligned at a pixel center (True) 
+           or if it is aligned at a pixel edge (False).
 
         """
-        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, torch.tensor(0.5, dtype = AP_config.ap_dtype, device = AP_config.ap_device), atol = 0.25)
+        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, 
+                            0.5, 
+                            atol = 0.25, 
+                            dtype = AP_config.ap_dtype, 
+                            device = AP_config.ap_device)
     
     @torch.no_grad()
-    def pixel_center_alignment(self):
+    def pixel_center_alignment(self)-> torch.Tensor:
         """
         Determine the relative position of the center of a pixel with respect to the origin (mod 1)
         """
         return ((self.origin + 0.5*self.pixelscale)/self.pixelscale) % 1
 
     @property
-    def data(self):
+    def data(self) -> torch.Tensor:
+        """
+        Returns the image data.
+        """
         return self._data
     @data.setter
-    def data(self, data):
+    def data(self, data)-> None:
+        """Set the image data."""
         self.set_data(data)
         
-    def set_data(self, data, require_shape = True):
+    def set_data(self, data: Union[torch.Tensor, np.ndarray], require_shape: bool = True):
+        """
+        Set the image data.
+
+        Args:
+            data (torch.Tensor or numpy.ndarray): The image data.
+            require_shape (bool): Whether to check that the shape of the data is the same as the current data.
+            
+        Raises:
+            AssertionError: If `require_shape` is `True` and the shape of the data is different from the current data.
+        """
         if self._data is not None and require_shape:
             assert data.shape == self._data.shape
-        self._data = data.to(dtype = AP_config.ap_dtype, device = AP_config.ap_device) if isinstance(data, torch.Tensor) else torch.as_tensor(data, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
-        
+        if isinstance(data, torch.Tensor):
+            self.data = data.to(dtype = AP_config.ap_dtype, device=AP_config.ap_device)
```

Comment:
This is where the recursion is happening. Be careful to use `self._data` not `self.data` when setting the value